Consistently applying Thread.currentThread().interrupt().

sduskis opened the pull request:
All InterruptedException handling should include Thread.currentThread().interrupt() as per:
http://stackoverflow.com/questions/4906799/why-invoke-thread-currentthread-interrupt-when-catch-any-interruptexception

igorbernstein2 commented:
Should this be added to the wider `catch(Exception|Throwable)` cases as well?

* https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BatchExecutor.java#L248

* https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java#L374

* https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java#L371

* https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java#L374

sduskis commented:
Great feedback! 

- BatchExecutor.java#L248 needs it.  Added it.
- BigtableBufferedMutator.java#L374 seems to not need it.  It uses a slight different path; the BigtableBufferedMutator.offer() method is the one that has the method that's prone to interruption.
- BulkMutation.java#L371 needs it.  Added it. 
- RowMerger.java#L374 does not need it.  There isn't any waiting involved.

PTAL

igorbernstein2 commented:
LGTM
